displaymanager.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import tdl
+
+from misc import Singleton, Vector
+from dungeon import Dungeon
+
+
+class Colors:
+    RED = (255, 0, 0)
+    GREEN = (0, 255, 0)
+    BLUE = (0, 0, 255)
+    BLACK = (0, 0, 0)
+    WHITE = (255, 255, 255)
+    WALL_DARK = (100, 100, 100)
+    GROUND_DARK = (120, 120, 120)
+    WALL_VISIBLE = (160, 160, 160)
+    GROUND_VISIBLE = (190, 190, 190)
+
+
+class DisplayManager(metaclass=Singleton):
+
+    """
+    Handles the display of entities and objects on the map.
+    """
+    # Constants
+    # A lot of these are just hardcoded values for debugging purposes.
+    # Ideally, they would be changeable by in-game settings.
+    SCREEN_WIDTH = 80
+    SCREEN_HEIGHT = 50
+    MAP_WIDTH = 80
+    MAP_HEIGHT = 44
+    GAME_TITLE = "Tonzo Studios Roguelike"
+    MAX_ROOMS = 30
+    MIN_ROOM_SIZE = 6
+    MAX_ROOM_SIZE = 10
+    MAX_ENTITIES_PER_ROOM = 3
+    FOV_ALGORITHM = "BASIC"
+    FOV_LIGHT_WALLS = True
+    FOV_RADIUS = 10
+
+    tdl.set_font('lucida10x10_gs_tc.png', greyscale=True, altLayout=True)
+    console = tdl.Console(SCREEN_WIDTH, SCREEN_HEIGHT)
+    root_console = tdl.init(SCREEN_WIDTH, SCREEN_HEIGHT, title=GAME_TITLE,
+                            fullscreen=False)
+
+    cur_map = None
+    fov_recompute = True
+
+    def gen_map(cls, player):
+        game_map = Dungeon(cls.MAP_WIDTH, cls.MAP_HEIGHT)
```

Comment:
@monzo94 this isn't display logic, but it's whatever... should stay here until we can manage to create a config class of some sort which would hold the big constants above, once that's done we can do the map generation in map and pass the map and the player to the display manager through an init() function or somesuch

Reply:
yes, this definitely shouldn't be here